detox/android/detox/src/main/java/com/wix/detox/DetoxInstrumentsManager.java

```diff
@@ -0,0 +1,141 @@
+            methodEventEndInterval = null;
+            methodEventMark = null;
+        }
+        hasProfiler = methodGetInstanceOfProfiler != null;
+    }
+
+    private final Context context;
+    private Object dtxProfilerInstance;
+
+    DetoxInstrumentsManager(Context context) {
+        this.context = context;
+    }
+
+    void startRecordingAtLocalPath(String path) {
+        if (!hasProfiler) {
+            return;
+        }
+        if (dtxProfilerInstance == null) {
+            dtxProfilerInstance = obtainProfiler();
+        }
+        final Object configuration = newConfiguration(path);
+        try {
+            methodStartRecording.invoke(dtxProfilerInstance, context, configuration);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    void stopRecording() {
+        if (!hasProfiler) {
+            return;
+        }
+        if (dtxProfilerInstance == null) {
+            return;
+        }
+        try {
+            methodStopRecording.invoke(dtxProfilerInstance);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    void eventBeginInterval(String category, String name, String id, String additionalInfo) {
+        if (dtxProfilerInstance == null) {
+            return;
+        }
+        try {
+            methodEventBeginInterval.invoke(dtxProfilerInstance, category, name, id, additionalInfo);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    void eventEndInterval(String id, String eventStatus, String additionalInfo) {
+        if (dtxProfilerInstance == null) {
+            return;
+        }
+        try {
+            methodEventEndInterval.invoke(dtxProfilerInstance, id, eventStatus, additionalInfo);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    void eventMark(String category, String name, String id, String eventStatus, String additionalInfo) {
+        if (dtxProfilerInstance == null) {
+            return;
+        }
+        try {
+            methodEventMark.invoke(dtxProfilerInstance, category, name, id, eventStatus, additionalInfo);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static Object newConfiguration(String path) {
+        try {
+            return constructorDtxProfilingConfiguration.newInstance(
+                    true,//recordPerformance
+                    500L,//samplingIntervalMillis
+                    new File(path),//recordingFile
+                    false//recordReactNativeTimersAsEvents
+            );
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private Object obtainProfiler() {
+        try {
+            return methodGetInstanceOfProfiler.invoke(null, context);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+}
```

Comment:
As in some of my previous comments, let's move some of this to a delegate class (e.g. `DetoxProfilerReflected`) and un-static stuff and then add some unit tests, plzzzz